### PR TITLE
feat : add update case study endpoint

### PR DESCRIPTION
**Pull request by rishusingh022**

### CODE REVIEW

File: src/controllers/case-study.controller.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const updateCaseStudyServices = require('../services/case-study.service');
+
+const updateCaseStudyController = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const { body } = req;
+    const updatedCaseStudy = await updateCaseStudyServices.updateCaseStudy(id, body);
+    if (!updatedCaseStudy) return res.status(404).json({ message: 'CaseStudy not found' });
```
Comment (rohini-senthil) [inline src/controllers/case-study.controller.js:8]: Can we change the error message to "Case study not found"

File: src/routes/user.route.js [hunk 2]
```diff
@@ -14,4 +14,5 @@ router.post('/users', authMiddlewares.reqAuthValidator, userControllers.createUs
 router.delete('/users/:id', authMiddlewares.reqAuthValidator, userControllers.deleteUser);
 router.put('/users/:id', updateIdValidator, updateBodyValidator, userControllers.updateUser);
```
Comment (rohini-senthil) [inline src/routes/user.route.js:15]: Can we add auth to this route?

File: src/routes/case-study.route.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const express = require('express');
+const { updateCaseStudyController } = require('../controllers/case-study.controller');
+const {
+  updateCaseStudyIdValidator,
+  updateCaseStudyBodyValidator,
+} = require('../middlewares/case-study.update.validator');
+
+const router = express.Router();
+
+router.put('/:id', updateCaseStudyIdValidator, updateCaseStudyBodyValidator, updateCaseStudyController);
```
Comment (rohini-senthil) [inline src/routes/case-study.route.js:10]: Can we add auth to this route?

File: src/middlewares/case-study.update.validator.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
+const Joi = require('joi');
```
Comment (rohini-senthil) [inline src/middlewares/case-study.update.validator.js:1]: Can we have all case study validators in the same file as `case-study.validator.js`, same for user as well

File: src/controllers/case-study.controller.js (hunk 1 above)
Comment (rishusingh022) [reply src/controllers/case-study.controller.js]: sure !

File: src/middlewares/case-study.update.validator.js (hunk 4 above)
Comment (rishusingh022) [reply src/middlewares/case-study.update.validator.js]: done

File: src/routes/case-study.route.js (hunk 3 above)
Comment (rishusingh022) [reply src/routes/case-study.route.js]: done

File: src/routes/user.route.js (hunk 2 above)
Comment (rishusingh022) [reply src/routes/user.route.js]: done

File: src/controllers/case-study.controller.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+const updateCaseStudyServices = require('../services/case-study.service');
```
Comment (Chetan-MckinseyBEBR) [inline src/controllers/case-study.controller.js:1]: Rename updateCaseStudyServices to updateCaseStudy. No need to specify service as part of variable name

File: src/middlewares/case-study.validator.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    if (error instanceof HttpError) {
+      res.status(error.status).json({ message: error.message });
+    } else {
+      res.status(500).json({ message: error.message });
+    }
+  }
+};
+
+module.exports = {
+  updateCaseStudyIdValidator,
```
Comment (Chetan-MckinseyBEBR) [inline src/middlewares/case-study.validator.js:56]: Why separate Validators? one for ID and one for body?

File: src/routes/case-study.route.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+const { updateCaseStudyIdValidator, updateCaseStudyBodyValidator } = require('../middlewares/case-study.validator');
+const authMiddlewares = require('../middlewares/request.validator');
+
+const router = express.Router();
+
+router.put(
+  '/:id',
+  authMiddlewares.reqAuthValidator,
+  updateCaseStudyIdValidator,
+  updateCaseStudyBodyValidator,
```
Comment (Chetan-MckinseyBEBR) [inline src/routes/case-study.route.js:12]: Can't we have a single validator?

**Review by Chetan-MckinseyBEBR — COMMENTED** [review]
Added few comments

File: tests/controllers/case-study.controller.test.js [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+        params: {
+          id: '1',
+        },
+        body: {
+          name: 'test',
+          description: 'test',
+          collaborators_ids: ['1', '2'],
+        },
+      };
+      const mockRes = {
```
Comment (rohini-senthil) [inline tests/controllers/case-study.controller.test.js:17]: Can we move all our mocks to a separate file under `tests/mocks` so it doesn't crowd the file

**Review by rohini-senthil — CHANGES_REQUESTED** [review]
Requested changes

File: src/middlewares/case-study.validator.js (hunk 6 above)
Comment (rishusingh022) [reply src/middlewares/case-study.validator.js:56]: they are separate because in delete we can use the same id validator to validator the id

Comment (rishusingh022) [reply src/middlewares/case-study.validator.js:56]: i have done that refactoring part in delete case study 
both the thing

File: tests/controllers/case-study.controller.test.js (hunk 8 above)
Comment (rishusingh022) [reply tests/controllers/case-study.controller.test.js:17]: sure !!

Comment (rishusingh022) [reply tests/controllers/case-study.controller.test.js:17]: creating that mock thing in the delete case study as it already has both things !!

File: src/services/case-study.service.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
+const { case_studies } = require('../models');
+
+const updateCaseStudy = async (id, caseStudy) => {
+  const updatedCaseStudy = await case_studies.findOne({ where: { case_study_id: id } });
```
Comment (rohini-senthil) [inline src/services/case-study.service.js:4]: This variable should be `caseStudy` since it's not updated yet

Comment (rohini-senthil) [reply src/services/case-study.service.js:4]: You can change the input name to something like `body`

**Review by rohini-senthil — APPROVED** [review]

**Pull request merged**